### PR TITLE
Add documentation for `group_by_use_nulls` setting

### PR DESCRIPTION
**Pull request by novikd**
### Changelog category (leave one):

- Documentation (changelog entry is not required)

### CODE REVIEW

File: docs/en/operations/settings/settings.md [hunk 1]
```diff
@@ -488,6 +488,23 @@ Possible values:
 
 Default value: 0.
 
+## group_by_use_nulls {#group_by_use_nulls}
+
+Changes the way [GROUP BY clause](../../sql-reference/statements/select/group-by.md) treats the types of aggregation keys.
```
Comment (DanRoscigno) [inline docs/en/operations/settings/settings.md:493]:
```suggestion
Changes the way the [GROUP BY clause](../../sql-reference/statements/select/group-by.md) treats the types of aggregation keys.
```

File: docs/en/operations/settings/settings.md [hunk 2]
```diff
@@ -488,6 +488,23 @@ Possible values:
...
+When `ROLLUP`, `CUBE` or `GROUPING SETS` are used, some aggregation keys may not be used to produce some result raws.
```
Comment (DanRoscigno) [inline docs/en/operations/settings/settings.md:494]:
```suggestion
When the `ROLLUP`, `CUBE`, or `GROUPING SETS` clauses are used, some aggregation keys may not be used to produce some result rows.
```

File: docs/en/operations/settings/settings.md [hunk 3]
```diff
@@ -488,6 +488,23 @@ Possible values:
...
+Columns for these keys are filled with either default value or `NULL` in corresponding raws depending on this setting.
```
Comment (DanRoscigno) [inline docs/en/operations/settings/settings.md:495]:
```suggestion
Columns for these keys are filled with either default value or `NULL` in corresponding rows depending on this setting.
```

File: docs/en/operations/settings/settings.md [hunk 4]
```diff
@@ -488,6 +488,23 @@ Possible values:
...
+
+Possible values:
+
+-   0 — The default value for the aggregation key type is used to produce missing values.
+-   1 — ClickHouse executes `GROUP BY` the same way as the SQL standard says. The types of aggregation keys are converted to [Nullable](../../sql-reference/data-types/nullable.md/#data_type-nullable). Columns for corresponding aggregation keys are filled with [NULL](../../sql-reference/syntax.md) for raws that didn't use it.
```
Comment (DanRoscigno) [inline docs/en/operations/settings/settings.md:500]:
```suggestion
-   1 — ClickHouse executes `GROUP BY` the same way as the SQL standard says. The types of aggregation keys are converted to [Nullable](/docs/en/sql-reference/data-types/nullable.md/#data_type-nullable). Columns for corresponding aggregation keys are filled with [NULL](/docs/en/sql-reference/syntax.md) for rows that didn't use it.
```

File: docs/en/operations/settings/settings.md [hunk 5]
```diff
@@ -488,6 +488,23 @@ Possible values:
...
+
+Default value: 0.
+
+See also:
+
+-   [GROUP BY clause](../../sql-reference/statements/select/group-by.md)
```
Comment (DanRoscigno) [inline docs/en/operations/settings/settings.md:506]:
```suggestion
-   [GROUP BY clause](/docs/en/sql-reference/statements/select/group-by.md)
```

File: docs/en/sql-reference/statements/select/group-by.md [hunk 6]
```diff
@@ -321,12 +328,12 @@ For every different key value encountered, `GROUP BY` calculates a set of aggreg
...
 This is the most general modifier.
-This modifier allows to manually specify several aggregation key sets (grouping sets).
-Aggregation is performed separately for each grouping set, after that all results are combined.
+This modifier allows manually specifying several aggregation key sets (grouping sets).
+Aggregation is performed separately for each grouping set, and after that, all results are combined.
 If a column is not presented in a grouping set, it's filled with a default value.
 
 In other words, modifiers described above can be represented via `GROUPING SETS`.
-Despite the fact that queries with `ROLLUP`, `CUBE` and `GROUPING SETS` modifiers are syntactically equal, they may have different performance.
+Despite the fact that queries with `ROLLUP`, `CUBE` and `GROUPING SETS` modifiers are syntactically equal, they may have different performances.
```
Comment (DanRoscigno) [inline docs/en/sql-reference/statements/select/group-by.md:336]:
```suggestion
Despite the fact that queries with `ROLLUP`, `CUBE` and `GROUPING SETS` modifiers are syntactically equal, they may perform differently.
```

**Review by DanRoscigno — APPROVED** [review]
Thanks!!

**Comment by DanRoscigno** [comment]
I changed all of the `../` relative links to `/docs/en/sql-reference` as we are seeing 404's when people refresh a page and then click on a link in the docs.  This seems to prevent the problem.

**Comment by novikd** [comment]
@DanRoscigno Thank you very much for the review. I applied all the fixes you suggested, but I did it only for the text I touched. Probably you want to change `../..` in all the files in another PR.

File: docs/en/operations/settings/settings.md [hunk 7]
```diff
@@ -488,6 +488,23 @@ Possible values:
...
+## group_by_use_nulls {#group_by_use_nulls}
+
+Changes the way the [GROUP BY clause](/docs/en/sql-reference/statements/select/group-by.md) treats the types of aggregation keys.
+When the `ROLLUP`, `CUBE`, or `GROUPING SETS` specifiers are used, some aggregation keys may not be used to produce some result rows.
+Columns for these keys are filled with either default value or `NULL` in corresponding rows depending on this setting.
+
+Possible values:
+
+-   0 — The default value for the aggregation key type is used to produce missing values.
+-   1 — ClickHouse executes `GROUP BY` the same way as the SQL standard says. The types of aggregation keys are converted to [Nullable](/docs/en/sql-reference/data-types/nullable.md/#data_type-nullable). Columns for corresponding aggregation keys are filled with [NULL](/docs/en/sql-reference/syntax.md) for rows that didn't use it.
```
Comment (serxa) [inline docs/en/operations/settings/settings.md:500]:
```suggestion
-   1 — ClickHouse executes `GROUP BY` the same way as the SQL standard says. The types of aggregation keys are converted to [Nullable](/docs/en/sql-reference/data-types/nullable.md). Columns for corresponding aggregation keys are filled with [NULL](/docs/en/sql-reference/syntax.md#null) for rows that didn't use it.
```

**Review by serxa — APPROVED** [review]
